packages/web/src/pages/pay-and-earn-page/components/TrackNameWithArtwork.tsx

```diff
@@ -1,21 +1,49 @@
-import { useGetTrackById } from '@audius/common/api'
-import { SquareSizes, statusIsNotFinalized } from '@audius/common/models'
+import {
+  useGetCurrentUserId,
+  useGetPlaylistById,
+  useGetTrackById
+} from '@audius/common/api'
+import {
+  SquareSizes,
+  Status,
+  USDCContentPurchaseType
+} from '@audius/common/models'
 import { Text } from '@audius/harmony'
 
 import DynamicImage from 'components/dynamic-image/DynamicImage'
+import { useCollectionCoverArt2 } from 'hooks/useCollectionCoverArt'
 import { useTrackCoverArt2 } from 'hooks/useTrackCoverArt'
 
 import styles from './TrackNameWithArtwork.module.css'
 
-export const TrackNameWithArtwork = ({ id }: { id: number }) => {
-  const { status, data: track } = useGetTrackById({ id })
-  const image = useTrackCoverArt2(id, SquareSizes.SIZE_150_BY_150)
-  const loading = statusIsNotFinalized(status) || !track
+export const TrackNameWithArtwork = ({
+  id,
+  contentType
+}: {
+  id: number
+  contentType: USDCContentPurchaseType
+}) => {
+  const isTrack = contentType === USDCContentPurchaseType.TRACK
+  const { status: trackStatus, data: track } = useGetTrackById(
+    { id },
+    { disabled: !isTrack }
+  )
+  const { data: currentUserId } = useGetCurrentUserId({})
```

Comment:
This feels a little fishy, passing empty object to it?

Reply:
Yeah... that's an audius-query problem. Would require some type-fu to get it to be optional I think, but I can look into it. I'll log a ticket in the audius-query-v2 project

Reply:
hm this is the same pattern we're using elsewhere, i don't have context on it tho, maybe @amendelsohn ?

Reply:
yeah you're doing it right. We'll have to fix it in audius-query. I made a ticket for it